src/GraphQL.Tests/Execution/DocumentExecuterTests.cs

```diff
@@ -50,6 +53,77 @@ public async Task Uses_ExecutionStrategySelector()
             mutationStrategy.Executed.ShouldBeTrue();
         }
 
+        [Fact]
+        public async Task TypedInterfaceMapping()
+        {
+            var services = new ServiceCollection();
+            services.AddGraphQL(b => b
+                .AddSchema<Schema1>()
+                .AddSchema<Schema2>()
+                .AddSystemTextJson());
+            services.AddSingleton(typeof(StringExecuter<>));
+            var provider = services.BuildServiceProvider();
+
+            // verify executing with Schema1 works with custom class
+            var executer1 = provider.GetRequiredService<StringExecuter<Schema1>>();
+            var result1 = await executer1.ExecuteAsync("{hero}");
+            result1.ShouldBe("{\"data\":{\"hero\":\"hello\"}}");
+
+            // verify executing with Schema2 works with IDocumentExecuter<> directly
+            var executer2 = provider.GetRequiredService<IDocumentExecuter<Schema2>>();
+            var result2 = await executer2.ExecuteAsync(new ExecutionOptions { Query = "{hero}", RequestServices = provider });
+            var serializer = provider.GetRequiredService<IGraphQLTextSerializer>();
+            serializer.Serialize(result2).ShouldBe("{\"data\":{\"hero\":\"hello2\"}}");
+
+            // verify that you cannot specify Schema with this implementation
+            await Should.ThrowAsync<InvalidOperationException>(async () => await executer2.ExecuteAsync(new ExecutionOptions { Schema = new Schema1(provider), Query = "{hero}", RequestServices = provider }));
```

Comment:
I would test message text here. As I said earlier it helps to understand the reason of exception.

Reply:
```suggestion
            var err = await Should.ThrowAsync<InvalidOperationException>(async () => await executer2.ExecuteAsync(new ExecutionOptions { Schema = new Schema1(provider), Query = "{hero}", RequestServices = provider }));
            err.Message.ShouldBe("ExecutionOptions.Schema must be null when calling this typed IDocumentExecuter<> implementation; it will be pulled from the dependency injection provider.");
```